src/learner.cc

```diff
@@ -691,6 +730,17 @@ class LearnerIO : public LearnerConfiguration {
     for (auto const& kv : attributes_) {
       learner["attributes"][kv.first] = String(kv.second);
     }
+
+    learner["feature_names"] = Array();
+    auto& feature_names = get<Array>(learner["feature_names"]);
```

Comment:
Any particular reason to re-fetch the `Array` we just created as opposed to keeping a reference to it at creation?

Reply:
Scratch that! Was thinking about this incorrectly. I see why.